ehr/src/org/labkey/ehr/utils/TriggerScriptHelper.java

```diff
@@ -1186,10 +1190,28 @@ private String checkOtherDrawsQuantity(String id, Date date, String rowObjectId,
 
         // Get records from the database in our date range that aren't part of the current transaction
         TableSelector tsdate = new TableSelector(ti, PageFlowUtil.set("objectid", "date", "quantity"), filter, null);
-        allBloods.addAll(tsdate.getArrayList(BloodInfo.class));
+        //get the db object
+            try (Results r = tsdate.getResults())
+            {
+
+                if (tsdate.getRowCount() > 0)
```

Comment:
Is this really the best pattern here? Could you call TableSelector.getArrayList() (or whatever that method is called)? At absolute minimum shouldnt you call tsdate.exists(), which is better than getRowCount(), prior to requesting a Results?

Reply:
To add to this: I would check whether just using a TableSelector.forEach() is sufficient. I am not sure there is any benefit to calling something like rowCount() or exists() first, followed by the query, since that is two separate SQL queries. It might be worth a small check on performance, but in general I would guess one query is better than two.

Reply:
I replaced this with a forEach()